internal/controllers/eventsource/controller.go

```diff
@@ -25,23 +28,47 @@ import (
 type EventSourceIngressController struct {
 	esLister      eslister.EventSourceLister
 	serviceLister corev1lister.ServiceLister
+	igc           v1.IngressConfigurator
 	config        EventSourceIngressControllerConfig
 }
 
 type EventSourceIngressControllerConfig struct {
-	gateway        types.NamespacedName
-	baseURL        string
-	adminNamespace string
+	Gateway        types.NamespacedName
+	BaseURL        string
+	AdminNamespace string
+	ipGetters      map[string]v1.IPGetter
```

Comment:
nit: it seems weird to have a "Config" struct with private fields.
Since this only gets set by EventSourceIngressController.SetIPGetter(), I think it can be moved into `EventSourceIngressController` struct